src/main/java/io/nats/client/impl/NatsPublishAck.java

```diff
@@ -38,7 +38,7 @@ public NatsPublishAck(byte[] response) throws IOException {
         }
 
         if (hasError()) {
-            throw new IllegalStateException(getError());
+            throw new JetStreamApiException(this);
```

Comment:
This way the error contains the full api response object and is consistent with other errors